### ISSUE TITLE
Can't override fixtures from conftest.py within subdirectory

### ISSUE THREAD

**Issue by jonahkagan** [schireson/pytest-alembic#17]
Hello, thanks for creating this awesome tool.

I was having some trouble overriding the `alembic_config` and `alembic_engine` fixtures in my tests. The docs don't specify how to go about overriding, so I was experimenting with anything I could think of.

My tests (as well as my `conftest.py` module) live within a subdirectory in my package structure. The only way I was able to get my fixtures to override the pytest-alembic fixtures was to move my `conftest.py` to the package root (the same directory I was running the `pytest` command from).

It seems like there's something funky going on with the ordering in which `pytest` loads fixtures from various places and plugins.

I was able to work around the problem by not running `pytest --test-alembic`, but rather just importing the test functions from pytest-alembic directly into a test file (where I overrode the fixtures). You can see what that looks like here: https://github.com/votingworks/arlo/pull/636.

It's actually kind of nice, because I get to specify which of the builtin tests I want to run (I don't have reversible migrations, so I had to skip the downgrading test). It made me think that perhaps making the tests more of a library to pull from instead of doing fancy magic to add them onto the end of the test run might actually make for a simpler user experience overall, so I thought I'd share.

In summary:
- It would be helpful to have documentation about overriding the fixtures
- I don't know why conftest.py had to be at the project root to override the fixtures - any ideas?
- Consider providing test functions as a library instead of a plugin

**Comment by DanCardin** [comment]
> It seems like there's something funky going on with the ordering in which pytest loads fixtures
I think the ordering problem is related to the location at which we're attaching the tests. In all our projects, the structure looks like
```
pytest.ini
src/
    code
tests/
    conftest.py
```
invoking pytest from the top level, and it picks up the conftest correctly.

In order to fix that part, we could accept an additional configurable pytest config option for declaring the root location of tests (inserted by `--test-alembic`).

> You can see what that looks like here: votingworks/arlo#636.
highly recommend https://pytest-alembic.readthedocs.io/en/latest/setup.html#pytest-mock-resources. Given your implementation of `alembic_engine`, it looks like you require postgres to already be running and you to be mutating its state with tests? Personally, I very much enjoy the UX of having clean docker containers spun up/down for me!

> because I get to specify which of the builtin tests I want to run 
If using `--test-alembic` was an option for you it **is** configurable using pytest config, but given the unresolved conftest issues, this is irrelevant to you.

> It made me think that perhaps making the tests more of a library
That's fair, I wanted a simpler default experience (though the requisite engine change to the default alembic.ini is already a bit sad). But I'd be happy to document its use as a library. Do you have any qualms/gotchas with the UX when running as a library? or is it just the lack of documentation?

**Comment by DanCardin** [comment]
I think I've addressed both the above points of my TL;DR, but lmk any more feedback and I can reopen if appropriate

**Comment by jonahkagan** [comment]
@DanCardin thanks so much for taking the time to respond and address this feedback. 

> highly recommend https://pytest-alembic.readthedocs.io/en/latest/setup.html#pytest-mock-resources
Yes, I checked it out and it looks useful. I may look into adding that separately, but that would be a change to our whole test infrastructure.

> Do you have any qualms/gotchas with the UX when running as a library? or is it just the lack of documentation?
No gotchas so far! Seems to be working fine. Thanks for adding the docs
